src/nunit.analyzers/ConstActualValueUsage/ConstActualValueUsageAnalyzer.cs

```diff
@@ -24,90 +22,74 @@ public class ConstActualValueUsageAnalyzer : BaseAssertionAnalyzer
-                       (argumentSymbol is IFieldSymbol fieldSymbol && fieldSymbol.IsConst);
-            }
-
-            bool IsStringEmpty(ExpressionSyntax expression)
+            static bool IsStringEmpty(IOperation operation)
             {
-                return expression is MemberAccessExpressionSyntax memberAccessExpression &&
-                    string.Equals(memberAccessExpression.Expression.ToString(), "string", StringComparison.OrdinalIgnoreCase) &&
-                    string.Equals(memberAccessExpression.Name.Identifier.Text, "Empty", StringComparison.Ordinal);
+                return operation is IFieldReferenceOperation propertyReference
+                    && propertyReference.Field.Name == nameof(string.Empty)
+                    && propertyReference.Field.ContainingType.SpecialType == SpecialType.System_String;
             }
 
-            void Report(ExpressionSyntax expression)
+            void Report(IOperation operation)
             {
                 context.ReportDiagnostic(Diagnostic.Create(
                     descriptor,
-                    expression.GetLocation()));
+                    operation.Syntax.GetLocation()));
             }
 
-            var actualExpression = assertExpression.GetArgumentExpression(methodSymbol, NunitFrameworkConstants.NameOfActualParameter);
+            var actualOperation = assertOperation.GetArgumentOperation(NunitFrameworkConstants.NameOfActualParameter);
 
-            if (actualExpression == null)
+            if (actualOperation == null)
                 return;
 
-            if (IsLiteralExpression(actualExpression))
+            if (IsLiteralOperation(actualOperation))
             {
-                Report(actualExpression);
+                Report(actualOperation);
                 return;
             }
 
-            if (!IsConstant(actualExpression) && !IsStringEmpty(actualExpression))
+            if (!actualOperation.ConstantValue.HasValue && !IsStringEmpty(actualOperation))
                 return;
 
             // The actual expression is a constant field, check if expected is also constant
-            var expectedExpression = this.GetExpectedExpression(assertExpression, methodSymbol, context.SemanticModel);
+            var expectedExpression = this.GetExpectedExpression(assertOperation);
 
-            if (expectedExpression != null && !IsConstant(expectedExpression) && !IsStringEmpty(expectedExpression))
+            if (expectedExpression != null && !expectedExpression.ConstantValue.HasValue && !IsStringEmpty(expectedExpression))
             {
-                Report(actualExpression);
+                Report(actualOperation);
             }
         }
 
-        private ExpressionSyntax? GetExpectedExpression(InvocationExpressionSyntax assertExpression, IMethodSymbol methodSymbol, SemanticModel semanticModel)
+        private IOperation? GetExpectedExpression(IInvocationOperation assertOperation)
```

Comment:
Perhaps rename `GetExpectedExpression` to `GetExpectedOperation`?

Reply:
Updated